distage/distage-testkit-scalatest/src/test/scala/izumi/distage/testkit/distagesuite/DistageTestExample.scala

```diff
@@ -37,15 +39,60 @@ class DistageTestExampleBIO extends DistageBIOSpecScalatest[zio.IO] with Distage
 
 }
 
+object DistageTestExampleBase {
+  sealed trait SetExample {
+    def v: Int
+  }
+  final case class SetElement1(v: Int) extends SetExample
+  final case class SetElement2(v: Int) extends SetExample
+  final case class SetElement3(v: Int) extends SetExample
+}
+
 abstract class DistageTestExampleBase[F[_]: TagK](implicit F: DIEffect[F]) extends DistageSpecScalatest[F] with DistageMemoizeExample[F] {
 
   override protected def config: TestConfig = super.config.copy(
-      pluginSource = super.config.pluginSource ++ PluginSource(PluginConfig.cached(Seq("xxx")))
+      pluginSource = super.config.pluginSource ++ PluginSource(PluginConfig.cached(Seq("xxx"))) ++ new ModuleDef {
+        make[SetElement1].from(SetElement1(1))
+        make[SetElement2].from(SetElement2(2))
+        make[SetElement3].from(SetElement3(3))
+
+        many[SetExample]
+          .weak[SetElement1]
+          .weak[SetElement2]
+          .weak[SetElement3]
+
+        many[SetExample].named("set-id")
+          .weak[SetElement1]
+          .weak[SetElement2]
+          .weak[SetElement3]
+      }
     )
 
   val XXX_Whitebox_memoizedMockCache = new AtomicReference[MockCache[F]]
 
   "distage test custom runner" should {
+    "support weak sets" in {
+      (
+        set: Set[SetExample],
+        s1: SetElement1,
+        s2: SetElement2,
+        s3: SetElement3,
+      ) =>
+        Quirks.discard(s1, s2, s3)
+        F.maybeSuspend(assert(set.size == 3))
+    }
+
+    "support named weak sets" in {
+      (
+        set: Set[SetExample] @Id("set-id"),
+        s1: SetElement1,
+        s2: SetElement2,
+        s3: SetElement3,
+      )=>
+        Quirks.discard(s1, s2, s3)
+        F.maybeSuspend(assert(set.size == 3))
```

Comment:
Could you add a test that summons Just Set, without set elements, and checks that size = 0?

Reply:
Yup, sure.

Reply:
You may want tests for memoized sets + non-memoized sets and assert that their behavior is different wrt inclusion on weak references that are only summoned in *other* tests